spring-ai-model/src/main/java/org/springframework/ai/tool/function/FunctionToolCallback.java

```diff
@@ -62,12 +62,13 @@ public class FunctionToolCallback<I, O> implements ToolCallback {
 
 	private final Type toolInputType;
 
-	private final BiFunction<I, ToolContext, O> toolFunction;
+	private final BiFunction<I, @Nullable ToolContext, O> toolFunction;
```

Comment:
Should probably be `? extends @Nullable ToolContext` (flexible nullability).

Same below.